decisions/20221206-frontend-log-instrumentation.md

```diff
@@ -0,0 +1,20 @@
+# Frontend log instrumentation
+
+- Status: accepted
+- Deciders: Nuraghe team
+- Tags: logs, frontend
+
+## Context
+Right now we only have a remote logger implementation that pushes logs to the backend.
+We want to also log stacktraces but without source-maps it would be hard for those stacktraces
+to be actually useful/readable.
+
+## Decision
+Since our logger implementation provides an `extra` parameter for additional info to add to the log entry, we will:
+
+- make an effort to log clear and effective messages that are actually useful when read by a developer/technician
```

Comment:
suggestion: we should probably be a bit more helpful with some guidelines here

let's ask ourselves: What's the decision we are recording here? 

I'd argue we could use this ADR to commit to some "logging principle", I'll jot down something:
- add a log line when initiating a state mutation operation
- add a log line when concluding (with either success or failure) an asynchronous operation

And so on

We are also about to add logs for page navigations and outgoing requests, so we may also add those to the ADR

**Please note**: I am not sure these are the things should list, but just some food for thought

Reply:
So with the team we decided to have this ADR on HOW to log
We'll have a new ADR on WHAT to log in the near future as soon as we actually start logging the way we intend.
Thank you!